### PR TITLE
Http now returns HttpResponse class that contains all needed data

### PR DESCRIPTION
**Pull request by patriotyk**
Catch all Uptane exceptions

### CODE REVIEW

File: src/httpclient.h [hunk 1]
```diff
@@ -6,18 +6,31 @@
...
+      : body(body_in),
+        http_status_code(http_status_code_in),
+        curl_code(curl_code_in),
+        error_message(error_message_in) {}
+  std::string body;
+  unsigned int http_status_code;
+  CURLcode curl_code;
+  std::string error_message;
+  operator bool() const { return curl_code == CURLE_OK; }
+  bool operator!() const { return curl_code != CURLE_OK; }
```
Comment (cajun-rat) [inline src/httpclient.h]: What about merging 3 checks together here:

1) That the http request completed OK
2) That the status code was 200 (or 20x if a response isn't needed)
3) That the response is valid JSON?

The JSON parsing could then be moved into HttpResponse, and "Utils::parseJSON(http.get(server + path).body);" would become:

~~~
HttpReponse http_response = http.get(path)
if (http_response.Ok()) {
  Json::Value json = http_response.json();
} else {
  // Handle error
}
~~~

File: src/sotahttpclient.cc [hunk 2]
```diff
@@ -85,7 +84,7 @@ void SotaHttpClient::startInstall(const data::UpdateRequestId &InstallingUpdate)
 void SotaHttpClient::sendUpdateReport(data::UpdateReport update_report) {
   std::string url = core_url + "/mydevice/" + config.device.uuid;
   url += "/updates/" + update_report.update_id;
-  http->post(url, update_report.toJson()["operation_results"]);
+  http->post(url, Json::FastWriter().write(update_report.toJson()["operation_results"]));
```
Comment (cajun-rat) [inline src/sotahttpclient.cc]: Looking at our code base, the only thing we send via http->post and http->put is JSON.  What about putting the JSON serialization inside the HttpClient?

**Comment by patriotyk** [comment]
@cajun-rat I've updated PR with all Your suggestions except of validation of json, because not always we get json. So the solution might be to check the content-type  and parse response in case it is json.

**Comment by cajun-rat** [comment]
(rebased to master)

**Pull request merged**